src/ristretto/dalek_range_proof.rs

```diff
@@ -103,6 +99,12 @@ impl RangeProofService for DalekRangeProofService {
     fn range(&self) -> usize {
         self.range
     }
+}
+
+impl RewindableRangeProofService for DalekRangeProofService {
+    type K = RistrettoSecretKey;
+    type P = Vec<u8>;
```

Comment:
Think `Proof` is a better name
```suggestion
    type Proof = Vec<u8>;
```